deep_quoridor/src/agents/alphazero/nn_evaluator.py

```diff
@@ -111,9 +113,10 @@ def evaluate_batch(self, games: list[Quoridor]):
         start_time = time.time()
         hashes = [g.get_fast_hash() for g in games]
 
+        cached_entries = {h: self.cache[h] for h in hashes if h in self.cache}
```

Comment:
Is this in case some of the keys drop from the cache? seems redundant, except for that corner case where the cache size is smaller than the batch size, or if you had a bug in the eviction 😛